diff --git a/searx/webapp.py b/searx/webapp.py
index 23dd56e66..e92957f10 100755
--- a/searx/webapp.py
+++ b/searx/webapp.py
@@ -1168,14 +1168,9 @@ def stats_errors():
 
 @app.route('/metrics')
 def stats_open_metrics():
-    password = settings['general'].get("open_metrics")
-
-    if not (settings['general'].get("enable_metrics") and password):
+    if not settings['general'].get("enable_metrics"):
         return Response('open metrics is disabled', status=404, mimetype='text/plain')
 
-    if not sxng_request.authorization or sxng_request.authorization.password != password:
-        return Response('access forbidden', status=401, mimetype='text/plain')
-
     filtered_engines = dict(filter(lambda kv: sxng_request.preferences.validate_token(kv[1]), engines.items()))
 
     engine_stats = get_engines_stats(filtered_engines)
